libraries/node-core-library/src/JsonSchema.ts

```diff
@@ -242,16 +263,16 @@ export class JsonSchema {
       // Validate each schema in order.  We specifically do not supply them all together, because we want
       // to make sure that circular references will fail to validate.
       for (const collectedSchema of collectedSchemas) {
-        if (!newValidator.validateSchema(collectedSchema._schemaObject)) {
+        if (!validator.addSchema(collectedSchema._schemaObject)) {
           throw new Error(
             `Failed to validate schema "${collectedSchema.shortName}":` +
               os.EOL +
-              JsonSchema._formatErrorDetails(newValidator.getLastErrors())
+              JsonSchema._formatErrorDetails(validator.errors!)
```

Comment:
We know `errors` is defined in this case?

Reply:
yes however I noticed that I failed to notice that `ajv.addSchema` has a different return type as `ajv.validateSchema` (ajv.addSchema uses ajv.validateSchema under the hood). I just made a tweak in the condition statement to correctly assert the error here, which also remove the needs to add the exclamation mark here.